libs/ui/src/components/molecules/GatedInput/GatedInput.stories.tsx

```diff
@@ -0,0 +1,143 @@
+import { WrappedInput } from '../WrappedInput';
+import { WrappedInputSelect } from '../WrappedInputSelect';
+import { WrappedSelect } from '../WrappedSelect';
+import { WrappedSwitch } from '../WrappedSwitch';
+import { WrappedTextArea } from '../WrappedTextArea';
+
+export default {
+  title: 'Molecules/Form/GatedInput',
+  component: GatedInput,
+} as ComponentMeta<typeof GatedInput>;
+
+const Template: ComponentStory<typeof GatedInput> = (args) => {
+  const { watch } = useFormContext();
+  const value = watch();
+
+  return (
+    <div style={{ margin: '4rem' }}>
+      <GatedInput {...args} />
+      <H3>{value[args.id] || 'Input text will appear here'}</H3>
+    </div>
+  );
+};
+
+const defautExplanationText = (
+  <>
+    <ParXs>Explanation paragraph #1</ParXs>
+    <ParXs>Explanation paragraph #2</ParXs>
+  </>
+);
+
+export const FullGatedInput = Template.bind({});
+FullGatedInput.args = {
+  gatedId: 'gatedInput',
+  gateLabel: 'Gated Input',
+  id: 'example',
+  label: 'Input',
+  placeholder: 'placeholder',
+  helperText: 'Test the action/controls',
+  info: 'This is controlled by the info prop',
+  number: false,
+  address: false,
+  long: false,
+  full: false,
+  warning: undefined,
+  error: undefined,
+  success: undefined,
+  input: WrappedInput,
+  updateHelperMsg: (formValues) => `This is the new value: ${formValues['example']}`,
+  children: defautExplanationText,
+};
+
+export const FullGatedInputSelect = Template.bind({});
+FullGatedInputSelect.args = {
+  gatedId: 'gatedInput',
+  gateLabel: 'Gated Input',
+  id: 'inputId',
+  label: 'Wrapped Input Select',
+  selectId: 'selectID',
+  info: 'This is controlled by the info prop',
+  options: [
+    { value: `${3600 * 24}`, name: 'Days' },
+    { value: `${3600}`, name: 'Hours' },
+    { value: `${60}`, name: 'Minutes' },
+  ],
+  placeholder: 'placeholder',
+  helperText: 'Test the action/controls',
+  selectPlaceholder: '-period-',
+  input: WrappedInputSelect,
+  updateHelperMsg: (formValues) =>
+    `This is the new value: ${formValues['selectID'] ? formValues['inputId'] * formValues['selectID'] : 0} seconds`,
+  children: (
+    <ParXs style={{marginTop: '5px'}}>
+      The expiration Date includes Voting and Grace periods. Adjust the days or hours
+      to update the expiration
+    </ParXs>
+  ),
+};
+
+export const FullGatedSelect = Template.bind({});
+FullGatedSelect.args = {
+  gatedId: 'gatedInput',
+  gateLabel: 'Gated Input',
+  id: 'example',
+  label: 'Complete Select',
+  placeholder: 'placeholder',
+  helperText: 'Test the action/controls',
+  info: 'This is controlled by the info prop',
+  options: [
+    { value: 'selectValue1', name: 'Select Label 1' },
+    { value: 'selectValue2', name: 'Select Label 2' },
+  ],
+  long: false,
+  full: false,
+  warning: undefined,
+  error: undefined,
+  input: WrappedSelect,
+  updateHelperMsg: (formValues) => `This is the new value: ${formValues['example'] || ''}`,
```

Comment:
Let's shy away from using callbacks to describe how we want our field elements to behave. It makes serialising the lego data difficult.

Reply:
same concerns from above about how to update the explanation text

Reply:
I'd really like to not have a component that takes in a callback as a means to produce helper text. There's a good pattern we can use to get around this. DM me on Discord when you're available to pair on this.